unexpected EOF in windows_restart provider with azure_rm builder

- Packer v1.0.3
- Windows
- Debug log https://gist.github.com/chrisrpatterson/687248e31d04ec90c0c51c608aa0e945

Packer is crashing periodically with an unexpected EOF error when using the windows_restart provider with the azure_rm builder.

posting this here for my own notes. Trace on my build of packer:

```
==> azure-arm: Waiting for machine to restart...
2018/02/06 12:24:42 packer: 2018/02/06 12:24:42 Check if machine is rebooting...
2018/02/06 12:24:49 packer: Build 'azure-arm' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> azure-arm: unexpected EOF

==> Builds finished but no artifacts were created.
panic: runtime error: invalid memory address or nil pointer dereference
2018/02/06 12:24:49 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x12709e6]
2018/02/06 12:24:49 packer:
2018/02/06 12:24:49 packer: goroutine 69 [running]:
2018/02/06 12:24:49 packer: net/http.redirectBehavior(0x2da0115, 0x4, 0x0, 0xc4201a4800, 0x0, 0x0, 0x0)
2018/02/06 12:24:49 packer: 	/usr/local/go/src/net/http/client.go:417 +0x26
2018/02/06 12:24:49 packer: net/http.(*Client).Do(0xc4204ef410, 0xc4201a4800, 0x6, 0xc4203005c0, 0x20)
2018/02/06 12:24:49 packer: 	/usr/local/go/src/net/http/client.go:615 +0x318
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/vendor/github.com/masterzen/winrm.clientRequest.Post(0x3f1f300, 0xc420280c60, 0xc42042a780, 0xc42027cf60, 0x0, 0x0, 0x0, 0x0)
2018/02/06 12:24:49 packer: 	/Users/mwhooker/go/src/github.com/hashicorp/packer/vendor/github.com/masterzen/winrm/http.go:82 +0x2f6
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/vendor/github.com/masterzen/winrm.ClientNTLM.Post(0x3f1f300, 0xc420280c60, 0xc42042a780, 0xc42027cf60, 0x4ac18f0, 0xc4205f1970, 0x2daf7da, 0xc4204fbf00)
2018/02/06 12:24:49 packer: 	/Users/mwhooker/go/src/github.com/hashicorp/packer/vendor/github.com/masterzen/winrm/ntlm.go:22 +0x49
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/vendor/github.com/masterzen/winrm.(*ClientNTLM).Post(0xc420280c20, 0xc42042a780, 0xc42027cf60, 0x15caf03, 0xc4205f1938, 0xc4205f1940, 0x5)
2018/02/06 12:24:49 packer: 	<autogenerated>:1 +0x5d
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/vendor/github.com/masterzen/winrm.(*Client).sendRequest(0xc42042a780, 0xc42027cf60, 0xc42027cf60, 0xc42027cf60, 0xc4204eee10, 0x2952025edd0)
2018/02/06 12:24:49 packer: 	/Users/mwhooker/go/src/github.com/hashicorp/packer/vendor/github.com/masterzen/winrm/client.go:102 +0x43
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/vendor/github.com/masterzen/winrm.(*Client).CreateShell(0xc42042a780, 0x0, 0x0, 0x0)
2018/02/06 12:24:49 packer: 	/Users/mwhooker/go/src/github.com/hashicorp/packer/vendor/github.com/masterzen/winrm/client.go:81 +0xa3
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/communicator/winrm.(*Communicator).Start(0xc420500f20, 0xc420165f20, 0xc420338b40, 0x3f1fb00)
2018/02/06 12:24:49 packer: 	/Users/mwhooker/go/src/github.com/hashicorp/packer/communicator/winrm/communicator.go:78 +0x33
2018/02/06 12:24:49 packer: github.com/hashicorp/packer/packer/rpc.(*CommunicatorServer).Start(0xc420282cc0, 0xc420290080, 0xc42053a160, 0x0, 0x0)
2018/02/06 12:24:49 packer: 	/Users/mwhooker/go/src/github.com/hashicorp/packer/packer/rpc/communicator.go:263 +0x1fb
2018/02/06 12:24:49 packer: reflect.Value.call(0xc4205a6c00, 0xc42000e5a8, 0x13, 0x2d9ffc1, 0x4, 0xc42069af20, 0x3, 0x3, 0x0, 0x0, ...)
2018/02/06 12:24:49 packer: 	/usr/local/go/src/reflect/value.go:434 +0x906
2018/02/06 12:24:49 packer: reflect.Value.Call(0xc4205a6c00, 0xc42000e5a8, 0x13, 0xc4202dff20, 0x3, 0x3, 0xc420164420, 0x0, 0x0)
2018/02/06 12:24:49 packer: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2018/02/06 12:24:49 packer: net/rpc.(*service).call(0xc420742040, 0xc420452730, 0xc420529308, 0xc4203a8300, 0xc420291480, 0x2709c40, 0xc420290080, 0x16, 0x275e6c0, 0xc42053a160, ...)
2018/02/06 12:24:49 packer: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2018/02/06 12:24:49 packer: created by net/rpc.(*Server).ServeCodec
2018/02/06 12:24:49 packer: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
2018/02/06 12:24:49 [INFO] 0 bytes written for 'stderr'
2018/02/06 12:24:49 [INFO] 0 bytes written for 'stdout'
```

--- a/provisioner/windows-restart/provisioner.go
+++ b/provisioner/windows-restart/provisioner.go
@@ -114,6 +114,12 @@ var waitForRestart = func(p *Provisioner, comm packer.Communicator) error {
 	var cmd *packer.RemoteCmd
 	trycommand := TryCheckReboot
 	abortcommand := AbortReboot
+
+	// This sleep works around an azure/winrm bug. For more info see
+	// https://github.com/hashicorp/packer/issues/5257; we can remove the
+	// sleep when the underlying bug has been resolved.
+	time.Sleep(1 * time.Second)
+
 	// Stolen from Vagrant reboot checker
 	for {
 		log.Printf("Check if machine is rebooting...")

